### ISSUE TITLE
Zoltan2::PartitioningProblem::resetParameters does not work

### ISSUE THREAD

**Issue by kddevin** [trilinos/Trilinos#1014]
It appears that Zoltan2::PartitioningProblem::resetParameters does not work.  Looking at the code, I think it doesn't do anything.  It needs to reset the parameter list in the Environment, but I cannot yet see where that reassignment is done.

This problem is exposed in examples/geometric/rcb_C.cpp; currently,  Zoltan2_ENABLE_Experimental must be enabled, although I suspect that CMAKE flag is not really relevant to this test.  When we fix this bug, we should remove the Zoltan2_ENABLE_Experimental guard in CMakeLists.txt.

This error was exposed by recent CDASH failures on muir, which enable Zoltan2_ENABLE_Experimental.

http://testing.sandia.gov/cdash/testDetails.php?test=36324661&build=2708092

@trilinos/zoltan2

**Comment by mmwolf** [comment]
I think you mean Zoltan2::Problem::resetParameters.  Is this correct?

I think this is my fault since I removed code from the setupProblemEnvironment() function
to fix other problems.  In particular, I removed:

  try{
    env_ = rcp(new Environment(*params, Teuchos::DefaultComm<int>::getComm()));
  }
  Z2_FORWARD_EXCEPTIONS
  envConst_ = rcp_const_cast<const Environment>(env_);

I think we should be able to put this code into the resetParameters function and this will fix the problem.  However, we need to pass a different comm instead of defaultComm (pass the current
comm if possible).

I can try to fix this but won't get to it until next week.

**Comment by MicheldeMessieres** [comment]
This can be fixed by adding a resetParameters(params) method to Environment which will reset the parameters and validate them without changing the initial comm setting. I'll open a PR shortly with this fix. I will also remove the Cmake experimental check.

**Comment by mmwolf** [comment]
I think I have fixed this.  However, feel free to improve this.  I found this whole Environment usage somewhat clunky.

**Comment by MicheldeMessieres** [comment]
Sorry, didn't notice you had just made the commit.
This alternative should achieve the same result - I'll open a PR only if requested then:

In Environment I would have added:

```
void Environment::resetParameters(Teuchos::ParameterList &problemParams)
{
  params_ = problemParams;
  commitParameters();
}
```

Then in Problem resetParameters() we would first call:
`env_->resetParameters(*params);`

**Comment by mmwolf** [comment]
It looks like my fix has cleared up this error.  That being said Michel's solution is probably a better one since it makes Environment less clunky.  So I'm ok if we decide to go that route instead.

**Comment by kddevin** [comment]
@MicheldeMessieres We would like you to implement the changes above as suggested, if you have time.  Would these changes also help #1004?   Or could they be done easily at the same time as fixing #1004?  
@krcb

**Comment by MicheldeMessieres** [comment]
I'll add #1004 changes to remove constructors without a communicator and swap the the original resetParameters fix for the new fix.

**Comment by MicheldeMessieres** [comment]
@kddevin These changed have been implemented in PR #1061.
